### PR TITLE
Fix OpenAI max_tokens + stop parameters

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

While testing vision, it looks like OpenAI enforces stricter type checking on `max_tokens` and `stop`. This PR fixes it.

```
ERROR: Error running assistant: Error querying `openai:gpt-4o`: error=[model_error(retryable=, request_id=req_662f77bf5b85f3059114da915bdc4699false)] OpenAIError: [invalid_request_error] [{'type': 'int_type', 'loc': ('body', 'max_tokens'), 'msg': 'Input should be a valid integer', 'input': None}, {'type': 'string_type', 'loc': ('body', 'stop', 'str'), 'msg': 'Input should be a valid string', 'input': None}, {'type': 'list_type', 'loc': ('body', 'stop', 'list[str]'), 'msg': 'Input should be a valid list', 'input': None}, {'type': 'list_type', 'loc': ('body', 'stop', 'list[list[int]]'), 'msg': 'Input should be a valid list', 'input': None}] (code: multi_actions_error)
```

### CODE REVIEW

**Review by fontanierh — APPROVED** [review]
LGTM

File: core/src/providers/openai.rs [hunk 1]
```diff
@@ -633,6 +628,12 @@ pub async fn streamed_completion(
     if model_id.is_some() {
         body["model"] = json!(model_id);
     }
+    if let Some(mt) = max_tokens {
+        body["max_tokens"] = mt.into();
+    }
+    if !stop.is_empty() {
```
Comment (spolu) [inline core/src/providers/openai.rs:634]: Aren't we changing the behavior a bit here? we should do Some(mt) ... && mt != 0 no ?

Comment (flvndvd) [reply core/src/providers/openai.rs:634]: You mean for `max_tokens`?

Comment (spolu) [reply core/src/providers/openai.rs:634]: Ok my bad I guess is_empty does the right thing :+1:  nvm me

**Pull request merged**